src/features/courses/course-repository.ts

```diff
@@ -483,6 +529,38 @@ class CourseRepository {
             throw new WrappedError('Could not create StudentTopicQuestionOverride', e);
         }
     }
+
+    async extendTopicQuestionByUser(options: ExtendTopicQuestionForUserOptions): Promise<UpsertResult<StudentTopicQuestionOverride>> {
+        try {
+            console.log(options);
+            const found = await StudentTopicQuestionOverride.findOne({where: {...options.where, active: true}});
+            if (!found) {
+                const newExtension = await StudentTopicQuestionOverride.create({...options.where, ...options.updates}, {validate: true});
```

Comment:
Not really pr feedback but a topic for conversation
I used spread but realized that typescript starts treating the type as any, it doesn't actually check if it's a valid spread
I think in the future we should probably move away from using spread operator in typescript

Reply:
I don't think that's a good reason to avoid using it. There might be better support in later versions of Typescript that we could benefit from, but I don't think this is a good enough reason to avoid using it.

Reply:
The point of typescript is to protect you from using incorrect typing, using this you could pass a string as a number etc.
I feel like anything that circumvents typescript is a big problem as you add more and more distrust in the types
i.e. on the front end we don't check types from the api calls so dates come as string and caused bugs (the very thing typescript tries to avoid)

Not a now issue but we should talk about it further